xarray/backends/h5netcdf_.py

```diff
@@ -452,19 +452,20 @@ def open_datatree(
             filename_or_obj,
             group=group,
         )
+        # Check for a group and make it a parent if it exists
         if group:
             parent = NodePath("/") / NodePath(group)
         else:
             parent = NodePath("/")
 
         manager = store._manager
-        ds = open_dataset(store, **kwargs)
-        tree_root = DataTree.from_dict({str(parent): ds})
+
+        groups_dict = {}
```

Comment:
This `groups_dict` should be created by calling `self.open_groups`, to avoid code duplication.